core/src/main/scala/kafka/controller/ControllerChannelManager.scala

```diff
@@ -358,10 +360,14 @@ class RequestSendThread(val controllerId: Int,
       // one concurrent access case considering the producer of the queue:
       // an item is put to the queue right after the condition check below.
       // That behavior does not change correctness since the inserted item will be picked up in the next round
-      while (!queue.isEmpty && shouldContinueMerging) {
+      // Fixme: tmp fix for incident-1517. We tried to merge limited amount of requests to a single LiCombinedControlRequest
+      //  so that a single request size is not larger than max limit which is 100MB.
+      var counter = maxRequestCountToMerge
+      while (!queue.isEmpty && shouldContinueMerging && maxRequestCountToMerge > 0) {
```

Comment:
This is a do-nothing change because maxRequestCountToMerge is always > 0. You want counter in there.

Reply:
Haha good call! Thanks Greg!